File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/LanguageExercisesCS/UsingLinqStart/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace UsingLinqStart {
    internal class Student {
        public Student(string name, string subject, int year, double[] marks) {
            this.name = name;
            this.subject = subject;
            this.year = year;
            this.marks = marks;
        }
        public string Subject {
            get { return subject; }
        }
        public string Name {
            get { return name; }
        }
        public int Year {
            set { year = value; }
        }
        public double [] Marks {
            get { return marks; }
        }

        public double Average() {
            double total = 0;
            foreach (double mark in marks) {
                total += mark;
            }
            return total/marks.Length;
        }

        public override string ToString() {
            string msg = "{0} studying {1} in year {2}";
            return string.Format(msg, name, subject, year);
        }

        private string name;
        private string subject;
        private int year;
        private double[] marks;
    }
}

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/LanguageExercisesVB/ClassDeclarationsFinish/Student.vb
Public Class Student
    Public Sub New(ByVal name As String, ByVal subject As String, ByVal year As Integer, ByVal marks() As Double)
        _name = name
        _subject = subject
        _year = year
        _marks = marks
    End Sub
    Public Property Year() As Integer
        Get
            Return _year
        End Get
        Set(ByVal value As Integer)
            _year = value
        End Set
    End Property
    Public Function average() As Double
        Dim total As Double
        For Each mark As Double In _marks
            total += mark
        Next
        Return total / _marks.Length
    End Function
    Public Overrides Function ToString() As String
        Dim msg As String = "{0} studying {1} in year {2}"
        Return String.Format(msg, _name, _subject, _year)
    End Function
    Private _name As String
    Private _subject As String
    Private _year As Integer
    Private _marks() As Double
End Class
